### PR TITLE
feat: add VitePress integration

### LINKED ISSUE #355: add VitePress integration
### Clear and concise description of the problem

Finally VitePress has been released with the `buildEnd` hook: we can now create an integration here using the code in the VitePress PR: https://github.com/vuejs/vitepress/pull/709#issuecomment-1146139835

### Suggested solution

Just add a new integration for VitePress: add a factory function to create the `async buildEnd` hook.

I need to check the logic for this integration!.

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by userquin**
I need to change the docs

closes #355

### CODE REVIEW

File: src/integrations/vitepress/index.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { resolveConfig } from 'vite'
+import type { VitePWAOptions, VitePluginPWAAPI } from '../../types'
+import { VitePWA } from '../../index'
+
+export function createBuildEndHook(userOptions: Partial<VitePWAOptions>): (siteConfig: any) => Promise<void> {
+  return async (siteConfig: any) => {
+    // run first integration hook
+    await userOptions.integrationHook?.(siteConfig)
```
Comment (userquin) [inline src/integrations/vitepress/index.ts:8]: I'll remove that, ppl can do it.

Comment (userquin) [reply src/integrations/vitepress/index.ts:8]: @antfu Finally I have left this option included

File: src/integrations/vitepress/index.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    const viteConfig = await resolveConfig({
+      plugins: [VitePWA(userOptions)],
+    },
+    'build',
+    'production',
```
Comment (userquin) [inline src/integrations/vitepress/index.ts:13]: should these 2 options be configurable?

**Comment by antfu** [comment]
Can't we await Vite 3.1 for the object hook? I am still not a fan of having integration specific logic built in.

**Comment by userquin** [comment]
> Can't we await Vite 3.1 for the object hook? I am still not a fan of having integration-specific logic built in.

We need to do this way for SvelteKit and VitePress, both repos reject the integration.

For VitePress there is no race condition we only need to use the build end hook 

For SvelteKit we need both, the integration and the sequential hook.

An alternative can be split the repo similar to UnoCSS with separate integrations.

File: docs/.vitepress/config.ts [hunk 3]
```diff
@@ -227,6 +308,7 @@ function prepareSidebar(idx: number) {
 const ogUrl = 'https://vite-plugin-pwa.netlify.app/'
 const ogImage = 'https://vite-plugin-pwa.netlify.app/og-image.png'
 
+// @ts-expect-error avoid TS2321: Excessive stack depth comparing types
```
Comment (userquin) [inline docs/.vitepress/config.ts:311]: I need to remove this line, it seems on IntelliJ is not working but it works in VSCode.

Comment (userquin) [reply docs/.vitepress/config.ts:311]: ok, this error seems to appear and dissapear randomly, so we can keep it: https://discord.com/channels/325477692906536972/1012326745753530419/1012347386653319248

**Comment by userquin** [comment]
included in the integration repo

**Pull request closed without merging**